link_linux.go

```diff
@@ -1580,9 +1586,57 @@ func LinkDeserialize(hdr *unix.NlMsghdr, m []byte) (Link, error) {
 	}
 	*link.Attrs() = base
 
+	// If the tuntap attributes are not updated by netlink due to
+	// an older driver, use sysfs
+	if link != nil && linkType == "tun" {
+		tuntap := link.(*Tuntap)
+
+		if tuntap.Mode == 0 {
+			ifname := tuntap.Attrs().Name
+			if flags, err := readSysPropAsInt64(ifname, "tun_flags"); err == nil {
+
+				if (flags & unix.IFF_TUN) != 0 {
+					tuntap.Mode = unix.IFF_TUN
+				} else if (flags & unix.IFF_TAP) != 0 {
+					tuntap.Mode = unix.IFF_TAP
+				}
+
+				if flags&unix.IFF_PERSIST == 0 {
```

Comment:
please `go fmt` this.

Reply:
Why here `if A & B == 0` and above `if (A&B) != 0` ?
Please be consistent, if parenthesis are not needed, do not add them